src/frameworks/angular/index.ts

```diff
@@ -53,35 +52,20 @@ export async function init(setup: any, config: any) {
 }
 
 export async function build(dir: string): Promise<BuildResult> {
-  const { targetStringFromTarget } = relativeRequire(dir, "@angular-devkit/architect");
-  const { architect, browserTarget, prerenderTarget, serverTarget } = await getContext(dir);
-
-  const scheduleTarget = async (target: Target) => {
-    const run = await architect.scheduleTarget(target, undefined);
-    const { success, error } = await run.output.toPromise();
-    if (!success) throw new Error(error);
-  };
-
+  const { targets, serverTarget, ngOptimizedImage, locales } = await getBuildConfig(dir);
   await warnIfCustomBuildScript(dir, name, DEFAULT_BUILD_SCRIPT);
-
-  if (!browserTarget) throw new Error("No build target...");
-
-  if (prerenderTarget) {
+  for (const target of targets) {
     // TODO there is a bug here. Spawn for now.
     // await scheduleTarget(prerenderTarget);
     const cli = getNodeModuleBin("ng", dir);
-    execSync(`${cli} run ${targetStringFromTarget(prerenderTarget)}`, {
+    spawnSync(cli, ["run", target], {
       cwd: dir,
       stdio: "inherit",
     });
-  } else {
-    await scheduleTarget(browserTarget);
-    if (serverTarget) await scheduleTarget(serverTarget);
   }
-
-  const wantsBackend = !!serverTarget;
-
-  return { wantsBackend };
+  const wantsBackend = !!serverTarget || ngOptimizedImage;
```

Comment:
maybe we could log the reasons for backend as we do with Next.js?

Reply:
absolutely, i expect more reasons as Angular is on a roll lately